### PR TITLE
Split program id from snapshot uuid

### PR DESCRIPTION
**Pull request by kasperl**

### CODE REVIEW

File: tools/firmware.toit [hunk 1]
```diff
@@ -858,9 +827,12 @@ class ImageHeader:
   flags= value/int -> none:
     header_[METADATA_OFFSET_] = value
 
-  program_id -> uuid.Uuid:
+  id -> uuid.Uuid:
     return read_uuid_ ID_OFFSET_
 
+  snapshot_uuid -> uuid.Uuid:
+    return read_uuid_ 0 // TODO(kasper): Wrong.
```
Comment (kasperl) [inline tools/firmware.toit:834]: Let me fix this before landing.

Comment (kasperl) [reply tools/firmware.toit]: Done.

File: tools/firmware.toit [hunk 2]
```diff
@@ -635,35 +598,46 @@ extract_binary envelope/Envelope --config_encoded/ByteArray -> ByteArray:
       --if_present=: json.decode it
       --if_absent=: {:}
 
+  // Compute relocatable images for all the containers.
+  images := {:}
+  entries.do: | name/string content/ByteArray |
+    if name.starts_with "\$" or name.starts_with "+": continue.do
```
Comment (erikcorry) [inline tools/firmware.toit:604]: Add comment about what $ and + mean.

File: tools/firmware.toit [hunk 3]
```diff
@@ -833,18 +798,24 @@ class Envelope:
...
-  static MARKER_OFFSET_   ::= 0
-  static ID_OFFSET_       ::= 8
-  static METADATA_OFFSET_ ::= 24
-  static UUID_OFFSET_     ::= 32
-  static HEADER_SIZE_     ::= 48
+  static MARKER_OFFSET_        ::= 0
+  static ID_OFFSET_            ::= 8
+  static METADATA_OFFSET_      ::= 24
+  static UUID_OFFSET_          ::= 32
+  static SNAPSHOT_UUID_OFFSET_ ::= 48 + 7 * 2 * 4  // 7 tables and lists.
```
Comment (erikcorry) [inline tools/firmware.toit:817]: Comments here and in program.h to link them together.

**Review by erikcorry — APPROVED** [review]

File: tools/firmware.toit (hunk 2 above)
Comment (kasperl) [reply tools/firmware.toit:604]: Done.

File: tools/firmware.toit (hunk 3 above)
Comment (kasperl) [reply tools/firmware.toit:817]: Done.

**Pull request merged**